omod/src/main/webapp/resources/scripts/htmlForm.js

```diff
@@ -258,24 +276,50 @@
     // TODO: these methods (getEncounter*Date*) will have to be modified when/if we switch datepickers
     // TODO: could/should be generalized so as not to be datepicker dependent?
 
-    htmlForm.setEncounterStartDateRange = function(date) {
+    //Set encounterDate datepicker start date
+    htmlForm.setEncounterStartDateRange = function(date , handletimezones) {
         if (getField('encounterDate.value')) {
-            getField('encounterDate.value').datepicker('option', 'minDate', date);
+            if (handletimezones) {
+                var startDateClientTimezone = new Date(date)
+                getField('encounterDate.value').datepicker('option', 'minDate', startDateClientTimezone)
+            }else{
+                getField('encounterDate.value').datepicker('option', 'minDate',  splitDate(date));
+            }
         }
     };
 
-    htmlForm.setEncounterStopDateRange = function(date) {
+    //Set encounterDate datepicker stop date
+    htmlForm.setEncounterStopDateRange = function(date, handletimezones) {
+        //Set browser timezone
+        if(jq("#encounterDate").find(".hfe-timezone")){
+            jq("#encounterDate").find(".hfe-timezone").val(Intl.DateTimeFormat().resolvedOptions().timeZone)
+        }
         if (getField('encounterDate.value')) {
-            getField('encounterDate.value').datepicker('option', 'maxDate', date);
+            if (handletimezones) {
+                var stopDateClientTimezone = new Date(date)
+                getField('encounterDate.value').datepicker('option', 'maxDate', stopDateClientTimezone > new Date ? new Date : stopDateClientTimezone)
+            }else{
+                getField('encounterDate.value').datepicker('option', 'maxDate', splitDate(date));
```

Comment:
Do we need `splitDate()` when we are not dealing with time zones?

Reply:
Yes we do need it, before we had dateTime , and split the time so we get only the date, and that was done here on a gsp file: https://github.com/openmrs/openmrs-module-htmlformentryui/pull/40/files/d32c855c9cb7b7a2543548d962754d651b52c6a5#diff-9920bfbf71a8b9f88051301b5ce589794fa2de063329605ce35381698ee79285L68-L79

But now we are doing this split on the JS (client-side) with that splitDate function

Reply:
What happens when we do not call splitDate() when we are not dealing with time zones?

Reply:
@dkayiwa the splitDate() it's only used when we are **not** dealing with timezones, because, in that case, we only want the date itself, not the time or the timezone.
If we were dealing with timezones(GP true) we wanted the full ISO8601 date, which includes the timezone associated with that date.